src/Product/WCProductAdapter.php

```diff
@@ -1070,4 +1083,27 @@ protected function format_attribute( $value, $attribute_id ) {
 
 		return $value;
 	}
+
+	/**
+	 * Gets a custom attribute from a product
+	 *
+	 * @param string $attribute_name - The attribute name to get.
+	 * @return string|null The attribute value or null if no value is found
+	 */
+	protected function get_custom_attribute( $attribute_name ) {
+		$product = $this->get_wc_product();
+
+		$attribute_value = $product->get_attribute( $attribute_name );
+
+		if ( ! $attribute_value ) {
+			$attribute_value = $product->get_meta( $attribute_name );
+		}
+
+		// We only support scalar values.
+		if ( ! is_scalar( $attribute_value ) ) {
+			return '';
+		}
+
+		return explode( ' ' . WC_DELIMITER . ' ', strval( $attribute_value ) )[0];
```

Comment:
Since we are sharing this function for both `get_attribute` and `get_meta`. Only in the case of `get_attribute` will it add the surrounding spaces around `WC_DELIMITER`. Would it be better to handle that consistently?

```suggestion
		$values = explode( WC_DELIMITER, (string) $attribute_value );
		$values = array_filter( array_map( 'trim', $values ) );
		return empty( $values ) ? '' : $values[0];
```